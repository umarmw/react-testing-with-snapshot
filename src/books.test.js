import React from 'react';
import renderer from 'react-test-renderer';

import Books from './Books';

it('should render an empty message when no books', () => {
  const elem = renderer.create(<Books />).toJSON();
  expect(elem).toMatchSnapshot();
});

it('should render a single book', () => {
    const books = ['Harry Potter']
    const elem = renderer.create(<Books books={books} />).toJSON();
    expect(elem).toMatchSnapshot();
});

it('should verify if the title matches', () => {
    const books = ['Harry Potter']
    const elem = renderer.create(<Books books={books} />).root;
    expect(elem.findByType("p").props.children).toEqual('Harry Potter');
});

it('should have 3 books listed', () => {
    const books = ['Harry Potter', 'The Lord of the Rings', 'The City of Dreaming Books']
    const elem = renderer.create(<Books books={books} />).root;
    expect(elem.findByType("ol").children.length).toEqual(3);
});

it('should render multiple books', () => {
    const books = ['Harry Potter', 'The Lord of the Rings']
    const elem = renderer.create(<Books books={books} />).toJSON();
    expect(elem).toMatchSnapshot();
});