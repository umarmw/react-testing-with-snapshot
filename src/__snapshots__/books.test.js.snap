// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a single book 1`] = `
<p>
  Harry Potter
</p>
`;

exports[`should render an empty message when no books 1`] = `
<span>
  We didn't find any books.
</span>
`;

exports[`should render multiple books 1`] = `
<ol>
  <li>
    Harry Potter
  </li>
  <li>
    The Lord of the Rings
  </li>
</ol>
`;
